leabra7/events.py

```diff
@@ -173,11 +192,67 @@ def from_name(cls, freq_name: str) -> "Frequency":
 
 
 CycleFreq = Frequency(name="cycle", end_event_type=Cycle)
-TrialFreq = Frequency(name="trial", end_event_type=EndPlusPhase)
+TrialFreq = Frequency(name="trial", end_event_type=EndTrial)
 EpochFreq = Frequency(name="epoch", end_event_type=EndEpoch)
 BatchFreq = Frequency(name="batch", end_event_type=EndBatch)
 
 
+class PhaseType(Enum):
+    plus = 1
+    minus = 2
+    none = 3
+
+
+class Phase():
+    """Defines network phases.
+
+    Args:
+        begin_event_type: Event type that marks the beginning of a phase.
+        end_event_type: Event type that marks the end of a phase.
+
+    """
+    # Stores a reference to each created frequency object, keyed by name
+    registry: Dict[str, "Phase"] = {}
+
+    name: str
+    begin_event_type: Type[Event]
+    end_event_type: Type[Event]
+
+    def __init__(self, name: str, phase_type: PhaseType) -> None:
+        self.name = name
```

Comment:
optional: Consider using `Type.getClassName()` to avoid the name parameter (one less thing for users to mess up).